### DISCUSSION TITLE
AaC Model Validation

### DISCUSSION THREAD

**Discussion by jondavid-black** [DevOps-MBSE/AaC#52] [Ideas]
First off let me say this is about "validation" as in "V&V" not as in "validating the model conforms to the DSL spec".

I was talking with a friend who raised an interesting question...let's say you can model a system perfectly, how do you know it will achieve the desired outcomes in context?  Up till now I've been almost completely focused on modeling a system to meet some set of requirement and interface specification.  This gets us to verification.  What we haven't considered is validation.  Could we use the AaC constructs or extend them in some way to actually evaluate the system specified in AaC to understand if it meets real world needs.

Let's create a situation to highlight he modeling need.

You are the chief engineer at Mario Androoti Carting and Princess Toadstump has asked you to build her a race car.  You know a lot about carting and are excited about the opportunity, but apparently times are tight in the Toadstump kingdom so all you'll get is a Time & Meterials contract with no profit except for a percentage of the winnings if the cart places in the top three positions in each race.  The princess has given you a race car specification in great detail, and you're confident you can model it, build it, test it, and deliver it.  But will you make any money at all if you do all that work?  How can you not only model the car you've been asked to build, but the real world context in which it must perform?  Then how can you analyze the car to understand how it may perform in that context?

There are already some useful building blocks in AaC.  First you can model your car and it's interactions.  Let's say you know enough about the other cars that you could build a "good enough" model for them as well since they have to conform to the same racing regulations.  You can even build up use cases around specific sequences of interaction to understand more about the flows in the system.  Modeling the environment may be something akin to modeling a race track.  It would have some number or corners in some sequence with some distance between them.  You would need a way to model the performance of your car within the environment.  For instance if you know your mass and your power levels maybe you could determine your acceleration and top speed for certain areas of the track.  Let's say there are a number of these measures of effectiveness you can define based on the properties of your car.  Now if you could apply the same or similar measures of effectiveness against the other cars, you may know something about your performance in the context of the track and other cars.  Could use use these measures of effectiveness in concert with some vignette of use cases to model a whole race and predict outcomes?  Are there aspects you could monte carlo and get some predictive statistics?

I thought it was an interesting problem.  It also highlights challenges with modeling a system to support analytical validation.  I really like the idea of doing the modeling up front before you build a thing that may look great on paper but wouldn't perform in the real world. 

I don't really want to take any action right now.  I did want to capture the thought for later consideration and give others a chance to think about it as well.

**Comment by lizzcondrey** [comment]
I am not finding any backlog issues or other discussions around this, so going with nothing more has come from this discussion? 

Sounds like it ish has come back up within the new validation type discussion #681 as we discussed how validators are really more verifiers of meeting system constraints and wanting to be more clear in that distinction?

**Reply by Coffee2Bits** [reply]
If I recall correctly this is more akin to Cameo's ability to simulate models in environments. This is a heavy lift, so we've not had the time or priority to address it.
